detection_test.py

```diff
@@ -242,6 +245,12 @@ def do_nms(k):
                 box = cls_box[valid_inds]
                 score = score[valid_inds]
                 det = np.concatenate((box, score.reshape(-1, 1)), axis=1).astype(np.float32)
+                # NOTE for set nms only
+                if bbox_xyxy.shape[0] == 2 * pRpn.proposal.post_nms_top_n:
```

Comment:
Better add a field in the config to directly indicate the set nms mode instead of depending on some hacks.